lib/puppet/file_system.rb

```diff
@@ -122,8 +122,8 @@ def self.each_line(path, &block)
   #
   # @api public
   #
-  def self.read(path)
-    @impl.read(assert_path(path))
+  def self.read(path, opts = {})
+    @impl.read(assert_path(path), opts)
```

Comment:
Agreed that Ruby's abstractions here are very weak compared to .NET as well (which has amazing encoding and locale support).

How would you like to surface a warning... via `Puppet.warning` or are you thinking of some other mechanism?